mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -72,97 +77,107 @@ public DefaultEntityFactory(ProcessorContext context) {
 
-      Map<Class<? extends Annotation>, Annotation> propertyAnnotations =
-          scanPropertyAnnotations(getMethod, field);
-      if (isTransient(propertyAnnotations, propertyName, transientProperties, getMethod, field)) {
-        continue;
-      }
+    // scan hierarchy for properties
+    for (TypeElement typeElement : typeHierarchy) {
+      for (Element child : typeElement.getEnclosedElements()) {
+        Set<Modifier> modifiers = child.getModifiers();
+        if (child.getKind() != ElementKind.METHOD
+            || modifiers.contains(Modifier.STATIC)
+            || modifiers.contains(Modifier.PRIVATE)) {
+          continue;
+        }
+        ExecutableElement getMethod = (ExecutableElement) child;
+        String getMethodName = getMethod.getSimpleName().toString();
+        if (!getMethodName.startsWith("get") || !getMethod.getParameters().isEmpty()) {
+          continue;
+        }
+        TypeMirror typeMirror = getMethod.getReturnType();
+        if (typeMirror.getKind() == TypeKind.VOID) {
+          continue;
+        }
+        String propertyName = Introspector.decapitalize(getMethodName.substring(3));
+        // skip properties we've already encountered.
+        if (encounteredPropertyNames.contains(propertyName)) {
+          continue;
+        }
 
-      int partitionKeyIndex = getPartitionKeyIndex(propertyAnnotations);
-      int clusteringColumnIndex = getClusteringColumnIndex(propertyAnnotations);
-      Optional<String> customCqlName = getCustomCqlName(propertyAnnotations);
-      Optional<String> computedFormula = getComputedFormula(propertyAnnotations, getMethod, field);
-
-      PropertyType propertyType = PropertyType.parse(typeMirror, context);
-      PropertyDefinition property =
-          new DefaultPropertyDefinition(
-              propertyName,
-              customCqlName,
-              computedFormula,
-              getMethodName,
-              setMethodName,
-              propertyType,
-              cqlNameGenerator);
-
-      if (partitionKeyIndex >= 0) {
-        PropertyDefinition previous = partitionKey.putIfAbsent(partitionKeyIndex, property);
-        if (previous != null) {
-          context
-              .getMessager()
-              .error(
-                  getMethod,
-                  "Duplicate partition key index: if multiple properties are annotated "
-                      + "with @%s, the annotation must be parameterized with an integer "
-                      + "indicating the position. Found duplicate index %d for %s and %s.",
-                  PartitionKey.class.getSimpleName(),
-                  partitionKeyIndex,
-                  previous.getGetterName(),
-                  property.getGetterName());
+        String setMethodName = getMethodName.replaceFirst("get", "set");
+        ExecutableElement setMethod = findSetMethod(typeElement, setMethodName, typeMirror);
+        if (setMethod == null) {
+          continue; // must have both
```

Comment:
This won't detect a property if the getter and setter come from different types in the hierarchy. That might be a fair requirement, but on the other hand it's pretty straightforward to scan the whole hierarchy for the setter, like you did for the field.

Reply:
Sure, I can add that.   I figured that `get` and `set` would be defined on the same type, but it's fair to define a super interface that only has getters.